code/core/src/androidTest/java/com/adobe/marketing/mobile/services/ui/message/views/MessageScreenTests.kt

```diff
@@ -330,7 +1032,16 @@ class MessageScreenTests {
         composeTestRule.onNodeWithTag(MessageTestTags.MESSAGE_CONTENT).assertExists().assertIsDisplayed()
         if (withBackdrop) {
             composeTestRule.onNodeWithTag(MessageTestTags.MESSAGE_BACKDROP).assertExists().assertIsDisplayed()
-                .assertIsDisplayed()
+        } else {
+            composeTestRule.onNodeWithTag(MessageTestTags.MESSAGE_BACKDROP).assertDoesNotExist()
+        }
+    }
+
+    private fun validateMessageExistsWhenClipped(withBackdrop: Boolean) {
```

Comment:
Is `assertDisplayed`  `false` for _all_ clipped tests because the message is completely off the screen? Is there a way to adjust the offset so that we ensure it partially inside the content bounds?

Also, you can augment the clipped behavior to `validateMessageAppeared(withBackdrop: Boolean)` 
```
validateMessageAppeared(withBackdrop: Boolean, clipped: Boolean = false) {
  ...
        composeTestRule.onNodeWithTag(MessageTestTags.MESSAGE_FRAME).assertExists()
           .also { if (!clipped) it.assertIsDisplayed() }
 ...
}

```

Reply:
`assertIsDisplayed()` is false for views even if they are partially clipped. I got `Assert failed: The component is not displayed!` error. Hence this method to only check if they existed. Refactored to reuse the existing `validateMessageAppeared` method